metpy/calc/thermo.py

```diff
@@ -1482,15 +1482,23 @@ def _find_append_zero_crossings(x, y):
 
     """
     # Find and append crossings to the data
-    crossings = find_intersections(x[1:], y[1:], np.zeros_like(y[1:]) * y.units)
-    x = concatenate((x, crossings[0]))
+    if hasattr(x, 'units'):
```

Comment:
Is all of this manual unit handling necessary now that you're using the `log_x=True` option?